core/chaincode/exectransaction_test.go

```diff
@@ -192,6 +190,28 @@ func TestExecuteDeployTransaction(t *testing.T) {
 	closeListenerAndSleep(lis)
 }
 
+// Test deploy of a transaction
+func TestExecuteDeployTransaction(t *testing.T) {
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a GOPATH with multiple elements
+func TestGopathExecuteDeployTransaction(t *testing.T) {
+	// add a trailing slash to GOPATH
+	// and a couple of elements - it doesn't matter what they are
+	os.Setenv("GOPATH", os.Getenv("GOPATH") + string(os.PathSeparator) + string(os.PathListSeparator) + "/tmp/foo" + string(os.PathListSeparator) + "/tmp/bar")
+	fmt.Printf("set GOPATH to: \"%s\"\n", os.Getenv("GOPATH"))
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a chaincode over HTTP.
+func TestHTTPExecuteDeployTransaction(t *testing.T) {
+	// The chaincode used here cannot be from the hyperledger repo
+	// itself or it won't be downloaded because it will be found
+	// in GOPATH, which would defeat the test
+	executeDeployTransaction(t, "http://github.com/lehors/fabric/examples/chaincode/go/chaincode_example01")
```

Comment:
@ghaskins I totally agree with the undesirable aspect of this dependency and it's only reluctantly that I put it in. Unless we are ready to change the current logic of the software (and there was a discussion a while ago that seems to end with agreement not to change that - see issue #737) there is no way around having _some_ external dependency.

Given that, I feel that depending on my fork is pretty safe. I have no reason to remove that any time soon and as long as I'm involved in the project, which I expect to remain true for the foreseeable future, it will keep being sync'ed with the main repo.
I do think it is better to have to live with that dependency than not to have such a critical piece of code not part of CI.